test/jmap/mail/calendar/reply/calendar_event_maybe_method_test.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:jmap_dart_client/jmap/mail/calendar/properties/event_id.dart';
+import 'package:jmap_dart_client/jmap/mail/calendar/reply/calendar_event_maybe_method.dart';
+import 'package:jmap_dart_client/jmap/mail/calendar/reply/calendar_event_maybe_response.dart';
+
+void main() {
+  final baseOption = BaseOptions(method: 'POST');
+  final dio = Dio(baseOption)..options.baseUrl = 'http://domain.com/jmap';
+  final dioAdapter = DioAdapter(dio: dio);
+  final dioAdapterHeaders = {"accept": "application/json;jmapVersion=rfc-8621"};
+  final httpClient = HttpClient(dio);
+  final processingInvocation = ProcessingInvocation();
+  final requestBuilder = JmapRequestBuilder(httpClient, processingInvocation);
+  final accountId = AccountId(Id('123abc'));
+  final successBlobId = Id('abc123');
+  final failureBlobId = Id('def456');
+  final notFoundBlobId = Id('ghi789');
+  final blobIds = [successBlobId, failureBlobId, notFoundBlobId];
+  final methodCallId = MethodCallId('c0');
+
+  Map<String, dynamic> constructData(CalendarEventReplyMethod method) => {
+    "using": method.requiredCapabilities
+      .map((capability) => capability.value.toString())
+      .toList(),
+    "methodCalls": [
+      [
+        method.methodName.value,
+        {
+          "accountId": accountId.id.value,
+          "blobIds": blobIds.map((id) => id.value).toList(),
+        },
+        methodCallId.value
+      ]
+    ]
+  };
+
+  Map<String, dynamic> constructResponse(CalendarEventReplyMethod method) => {
+    "sessionState": "abcdefghij",
+    "methodResponses": [[
+      method.methodName.value,
+      {
+        "accountId": accountId.id.value,
+        "maybe": [successBlobId.value],
+        "notMaybe": [failureBlobId.value],
+        "notFound": [notFoundBlobId.value],
+      },
+      methodCallId.value
+    ]]
+  };
+
+  group('calendar event maybe method', () {
+    final method = CalendarEventMaybeMethod(accountId, blobIds: blobIds);
+
+    test('should succeed with success blob data, '
+    'and fail with failure blob data '
+    'and not found with not found blob data', () async {
+      // arrange
+      final invocation = requestBuilder.invocation(method, methodCallId: methodCallId);
+      dioAdapter.onPost(
+        '',
+        (server) => server.reply(200, constructResponse(method)),
+        data: constructData(method),
+        headers: dioAdapterHeaders,
+      );
+
+      // act
+      final response = (await (requestBuilder..usings(method.requiredCapabilities))
+        .build()
+        .execute())
+          .parse<CalendarEventReplyResponse>(
+            invocation.methodCallId,
+            CalendarEventMaybeResponse.deserialize); 
+      
+      // assert
+      expect(
+        (response as CalendarEventMaybeResponse?)?.maybe,
+        equals([EventId(successBlobId.value)]));
+      expect(response?.notMaybe, equals([failureBlobId]));
+      expect(response?.notFound, equals([notFoundBlobId]));
```

Comment:
Please separate test cases: 
- Success
- Failure
- Both Success and Failure

Reply:
I don't think there's a need for that since we can bulk request and get bulk response, all available response cases can be bundle into 1 request.

Reply:
That's not unit testing, your thinking is wrong. We must specifically test cases to avoid unwanted situations from occurring. Are you sure you won't let anything slip through?

Reply:
> Please separate test cases:
> 
>     * Success
> 
>     * Failure
> 
>     * Both Success and Failure

Every single cases you listed here are covered in that one test. Response from the server are LIST of values, not single value.


> That's not unit testing, your thinking is wrong. We must specifically test cases to avoid unwanted situations from occurring. Are you sure you won't let anything slip through?

Then yes, I'm sure there's nothing can slip through this test.